types/alpinejs/index.d.ts

```diff
@@ -103,10 +103,10 @@ export interface InterceptorObject<T> {
 }
 
 type InferInterceptor<T> = T extends InterceptorObject<infer U> ? U
-    : T extends Record<string | number | symbol, unknown> ? {
-            [K in keyof T]: InferInterceptor<T[K]>;
-        }
-    : T;
+    : keyof T extends never ? T
+    : {
+        [K in keyof T]: InferInterceptor<T[K]>;
+    };
```

Comment:
@ekwoka I noticed that this change doesn't let me store complex object in component.
If install the version @types/alpinejs@3.13.5, this problem goes away:
![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/8982274/7435d239-d030-40c7-babd-0a64a4c410f7)

Reply:
😭 

If it wasn't for interceptors being how they are (I didn't make them don't blame me), this would be so easy...

Maybe, for now, this can just drop support for nested interceptors....

but then I use nested interceptors in projects....maybe limit the depth a certain amount?...

To at least reduce the potential conflicts...

I'll try to get this figured out.

Do you have like the full TS error? Like where is the spot that it determines as being "not assignable"? I assume it's because there is some kind of `any` somewhere....

Reply:
#69050 should solve this (or at least did In my checks...)

Reply:
I just tested on my side, it's solved!
Thank you so much for this quick fix, you rock! 💯